BetterHolidays/markets/nyse.py

```
@@ -1,180 +1,251 @@
+        table: "elm.Table" = doc.inner_html.get_elements_by_class_name("table-data")[0]
+
+        table_dict = table.to_dict()
+        holidays = table_dict.pop("Holiday")
+
+        for year, dates in table_dict.items():
+
+            def handle_date(date: "str"):
+                split_date = date.split(" ")
+
+                return dt.date(
+                    int(year),
```

Ruff (0.8.2)

Function definition does not bind loop variable `year`

(B023)